### ISSUE TITLE
Relitigate ignoring of pre-initial dependence.

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc#1544]
(This arose as a side-issue of #1540 which argues against pre-initial-ignore specifically in warm starts).

Ignoring pre-initial dependence was invented as a workaround for a deficiency of cylc-5 that no longer exists.  E.g. this suite would stall at start-up if we did not ignore `foo[2014]` (and if we didn't use old-style *cold-start tasks* and a conditional trigger):
```
[scheduling]
    initial cycle point = 2015
    [[dependencies]]
        [[[R1]]] # first cycle point
            graph = start => foo
        [[[P1Y]]]  # every cycle point (including the first)
            graph = foo[-P1Y] => foo
```
Since cylc-6 though, we have the ability to handle any start-up scenario correctly without ignoring pre-initial dependence:
```
[scheduling]
    initial cycle point = 2015
    [[dependencies]]
        [[[R1]]] # first cycle point
            graph = start => foo
        [[[R/+P1Y/P1Y]]]  # second and subsequent points
            graph = foo[-P1Y] => foo
```
This second version is only harder to understand than the first if you don't understand our graph syntax.  Arguably it is easier to understand because (a) there is no overlap between the R1 and general R sections; and (b) the graph syntax alone is sufficient to describe exactly what happens at start-up.  In the first case, something mysterious has to happen to allow the suite to start up (i.e. pre-initial dependence is automatically ignored). 

In my opinion we should deprecate this behaviour.  It is now unnecessary, it is quite complicated code, and as a workaround it is liable to have unforeseen consequences in edge cases (e.g. #1540 for its application to warm starts).

**Comment by hjoliver** [comment]
After some more thought on this, I think the default behaviour should be just like a warm-start (according to #1540 that is): at start-up, any pre-initial dependence should be retained and the pre-initial tasks should be inserted in the 'held' state.  Such a suite will stall at start-up, but the reason why will now be very obvious: the initial tasks depend on pre-initial tasks, and something needs to be done about that.  To proceed, users can either:
  * (ideally) fix their suite graph structure as in the second example above, or
  * manually set the pre-initial tasks to 'succeeded' (or other states if necessary)
  * optionally start the suite with any pre-initial tasks already succeeded (or whatever) as a proxy for the current pre-initial-ignore behaviour - like my proposed warm start option in #1540). 
  * OR perhaps we could even retain the current pre-initial ignore behaviour as on option.

We could retain current pre-initial handling as the default, but in principle I'd prefer not to because as I've argued above it is not really *the right thing to do*: it is magic behaviour that was invented to get around a deficiency of cylc-5 that no longer applies.

**Comment by hjoliver** [comment]
Finally, as an alternative to setting pre-initial tasks to 'held' in option 2 above, we could have a new task state for that purpose.

**Comment by hjoliver** [comment]
[meeting] with the pre-start warm start issue addressed (#1905) this can be closed as no further problems have been identified, pre-start ignore undoubtedly does make the start-up graph easier for users, and there's nothing to stop users doing this "properly" if they want to :grimacing:
